### PR TITLE
Move to ECS

### PR DESCRIPTION
**Pull request by davidrusu**
[WIP]

### CODE REVIEW

File: src/Utilities/MakeJavaSuckLess.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package Utilities;
+
+public class MakeJavaSuckLess {
```
Comment (ghost) [inline src/Utilities/MakeJavaSuckLess.java:3]: Please rename this class to give a better idea of what it does

File: src/gameengine/ecs/Entity.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    /**
+     * Fetches the component data for this entity.
+     * ie. If an entity has a PhysicalComponent, entity.getComponent(PhysicalComponent.class) will return the
+     * physical component associated with this entity.
+     */
+    public <T extends Component> T getComponent(Class<T> componentClass) {
+        Component component = componentMap.get(componentClass);
+        if (component == null) throw new EntityMissingComponentException(componentClass);
```
Comment (ghost) [inline src/gameengine/ecs/Entity.java:23]: Lets avoid exceptions as much as possible during regular engine runtime (but they are fine during initial setup / configuration)

File: src/gameengine/ecs/Scheduler.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package gameengine.ecs;
+
+public class Scheduler {
```
Comment (ghost) [inline src/gameengine/ecs/Scheduler.java:3]: Lets get rid of the scheduler for the simplistic first attempt

File: src/gameengine/ecs/System.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package gameengine.ecs;
+
+import java.util.List;
+
+public interface System {
```
Comment (ghost) [inline src/gameengine/ecs/System.java:5]: Please add methods entityCreated & entityDestroyed

File: src/gameengine/ecs/System.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+    Class[] requiredComponents();
+
+    Scheduler schedule();
+
+    void update(List<Entity> entities);
```
Comment (ghost) [inline src/gameengine/ecs/System.java:11]: The elapsed time is missing and I wonder if we need to also pass additional information like the context

File: src/gameengine/ecs/common/DiscreteCollisionSystem.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        // return Scheduler.beforeContextStart()
+        // return Scheduler.beforeContextExit()
+    }
+
+    public void update(List<Entity> entities) {
+        // this is where you do the stuff
+
+        for (int i = 0; i < entities.size(); i++) {
+            Entity a = entities.get(i);
+            PositionComponent aPos = a.getComponent(PositionComponent.class);
```
Comment (ghost) [inline src/gameengine/ecs/common/DiscreteCollisionSystem.java:42]: Hmm, using the class objects doesn't look as clean as I hoped.  Is there a better way?

**Comment by davidrusu** [comment]
No longer exploring ECS

**Pull request closed without merging**